model/SearchEngine/Driver/Elasticsearch/QueryBuilder.php

```diff
@@ -207,24 +214,66 @@ private function buildConditionsByType(string $type, array $blocks): array
         return $this->getResourceConditions($blocks);
     }
 
+    private function containsLogicalModifier(string $block) {
+        $found = false;
+        foreach(self::LOGIC_MODIFIERS as $key => $logicModifier) {
+            if(strpos($block, $logicModifier) !== false) {
+                $found = true;
+                break;
+            }    
+        }    
+        return $found;
+    }
+
+    private function buildLogicCondition(string $block) {
+        $condition = '';
+        if(strpos($block, self::LOGIC_MODIFIERS['and'])) {
+            $logicBlocks = preg_split('/( '.self::LOGIC_MODIFIERS['and'].' )/i', $block);
+            $conditions = array_map([$this, 'buildConditionFromTheBlock'], $logicBlocks);
+            $condition = sprintf('(%s)', implode(' AND ', $conditions));
+        } elseif (strpos($block, self::LOGIC_MODIFIERS['or'])) {
+            $logicBlocks = preg_split('/( '.self::LOGIC_MODIFIERS['or'].' )/i', $block);
+            $conditions = array_map([$this,'buildConditionFromTheBlock'], $logicBlocks);
+            $condition = sprintf('(%s)', implode(' OR ', $conditions));
+        } elseif (strpos($block, self::LOGIC_MODIFIERS['not'])) {
+            $logicBlocks = preg_split('/( ' . self::LOGIC_MODIFIERS['not'] . ' )/i', $block);
+            $conditions = array_map([$this,'buildConditionFromTheBlock'], $logicBlocks);
+            $condition = sprintf('NOT (%s)', implode(' OR ', $conditions));
+        }
+        return $condition;
+    }
+        
+
     private function getResourceConditions(array $blocks): array
     {
+        
         $conditions = [];
+
         foreach ($blocks as $block) {
-            $queryBlock = $this->parseBlock($block);
-
-            if (empty($queryBlock->getField())) {
-                $conditions[] = sprintf('("%s")', $queryBlock->getTerm());
-            } elseif ($this->isStandardField($queryBlock->getField())) {
-                $conditions[] = sprintf('(%s:"%s")', $queryBlock->getField(), $queryBlock->getTerm());
-            } else {
-                $conditions[] = $this->buildCustomConditions($queryBlock);
-            }
-        }
+            
+            if($this->containsLogicalModifier($block)) {
+                $conditions[] =  $this->buildLogicCondition($block);
+            }else{
+                $conditions[] =  $this->buildConditionFromTheBlock($block);
+            }    
+        }    
 
         return $conditions;
+    }    
+
+
+    private function buildConditionFromTheBlock(string $block) {
```

Comment:
chore suggestion: Please use earlier return as mentioned in other examples and use the proper spacing and return type in the method

Reply:
Thanks, fixed